m_etcd/coordinator.go

```diff
@@ -137,7 +138,7 @@ func NewEtcdCoordinator(nodeID, namespace string, hosts []string) (metafora.Coor
 
 	if nodeID == "" {
 		hn, _ := os.Hostname()
-		nodeID = hn + "-" + uuid.NewRandom().String()
+		nodeID = fmt.Sprintf("%s-%x", hn, rand.Int63())
```

Comment:
You need to seed the random id generator, to support two nodes on the same host (not in a test).   Testcases pass because they share the same default `rand` so it's technically two calls to `rand.Int63()`, but if you ran two instances on the same host they would create the same ID.   http://play.golang.org/p/WboRmHZUCd